src/Microsoft.OData.Core/UriParser/ODataPathInfo.cs

```diff
@@ -35,7 +35,10 @@ public ODataPathInfo(ODataPath odataPath)
                 }
 
                 this.targetNavigationSource = lastSegment.TargetEdmNavigationSource;
-                this.targetEdmType = lastSegment.TargetEdmType;
+
+                // If the last segment is a TypeSegment, we should assign the targetEdmType as lastSegment.EdmType
+                // instead of lastSegment.TargetEdmType which references the Base Type EdmType.

```

Comment:
Shouldn't the TargetEdmType of a TypeSegment be the EdmType?

Do we have other places in code where we make the same assumption (that the targetEdmType of a Type Segment is the actual EdmType, not the base type?

Reply:
Reverted this change since it was solving one use case `$expand=Prop1($expand=Prop2`.
Handling this in the `GenerateExpandItem` method in the `SelectExpandBinder`